### PR TITLE
Fix/jetpack footer inconsistent behavior

### LINKED ISSUE #31969: Jetpack Footers: inconsistent behavior
### Impacted plugin

Jetpack

### Quick summary

From #31627 @bindlegirl : 

The logged-in About link on the Jetpack dashboard (and other) pages shows up as an external link, although it's not. However, it loads the About page in a new tab, which differs from the Terms and Airline pages.

There are inconsistencies between the MyJetpack footer and other pages. This is what I see:

    Logged out About link on the Jetpack page will load https://jetpack.com/about/?site=nutritious-fairy.jurassic.ninja and MyJetpack https://jetpack.com/about
    Logged in About link on the Jetpack page will load /wp-admin/admin.php?page=jetpack_about and MyJetpack will load https://jetpack.com/about/

Privacy on connected MyJetpack is still an external link.

### Steps to reproduce

View the footers with the Jetpack admin and a My Jetpack page.

### A clear and concise description of what you expected to happen.

_No response_

### What actually happened

_No response_

### Impact

All

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by darssen**
Fixes #31969

This is a fallback from #31627 where some inconsistencies between the MyJetpack footer and other pages appeared.
## Proposed changes:

The changes focus on
- Make **Privacy** and **About** links in the footer non-external for connected scenarios.
- Add consistent behavior in the Footer of My Jetpack to the footers in the rest of the Jetpack pages.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Spin up a test site with this branch
- Make sure your site is not connected to Jetpack
- Visit the _Jetpack_ section
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `https://jetpack.com/about/`
  - Privacy: `https://automattic.com/privacy/`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `https://automattic.com/about/`
- Visit the _My Jetpack_ section
  - In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `https://jetpack.com/about/`
  - Privacy: `https://automattic.com/privacy/`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `https://automattic.com/about/`
- Connect your site to Jetpack
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `/wp-admin/admin.php?page=jetpack_about`
  - Privacy: `/wp-admin/admin.php?page=jetpack#/privacy`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `/wp-admin/admin.php?page=jetpack_about`
- Visit the _My Jetpack_ section
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `/wp-admin/admin.php?page=jetpack_about`
  - Privacy: `/wp-admin/admin.php?page=jetpack#/privacy`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `/wp-admin/admin.php?page=jetpack_about`
- Click on one of the Learn More buttons in one of the cards like VaultPress Backup
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `/wp-admin/admin.php?page=jetpack_about`
  - Privacy: `/wp-admin/admin.php?page=jetpack#/privacy`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `/wp-admin/admin.php?page=jetpack_about`

### CODE REVIEW

**Comment by darssen** [comment]
I don't like that the [getRedirectUrl](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/components/tools/jp-redirect/index.ts#L23) function adds the `site` parameter for all the Jetpack pages aside from My Jetpack. This makes things like redirecting to `tos` being redirected to https://wordpress.com/tos/?site=YOUR_SITE

Not sure about a good way to avoid it.

**Comment by darssen** [comment]
Also just realized that the initial change is only good for main My Jetpack page, but it is not working inside things like [page=my-jetpack#/add-backup](url). One possibility would be to add it to every place `AdminPage` is used. Another would be for `JetpackFooter` to don't need it as a prop, and use inside it maybe a global similar to the one I am using in `MyJetpackScreen` , the problem is that the `myJetpackInitialState` var does not belong outside MyJetpack, so still looking for a more elegant solution.

File: projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx [hunk 1]
```diff
@@ -120,7 +120,7 @@ export default function MyJetpackScreen() {
 	}
 
 	return (
-		<AdminPage>
+		<AdminPage siteAdminUrl={ window?.myJetpackInitialState?.adminUrl }>
```
Comment (andrii-lysenko) [inline projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx:123]: I'm wondering if we could pull off this information from some Redux store.

**Review by andrii-lysenko — COMMENTED** [review]
Overall, it seems reasonable to me. I agree with your comment and maybe we should consider incapsulate all logic inside the footer for consistent behavior.
Maybe, we could reuse some redux store for the latest state.

**Comment by darssen** [comment]
Thanks for the feedback @andrii-lysenko. 
After following some of the internal suggestions, I have updated the PR adding a tool inside` js-packages/components` to retrieve the `siteAdminUrl` and use it directly in the footer. It uses `window` var that gets populated from different Redux stores. Couldn't think of a more elegant solution.

File: projects/js-packages/components/tools/get-site-admin-url/index.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+declare global {
```
Comment (andrii-lysenko) [inline projects/js-packages/components/tools/get-site-admin-url/index.ts]: Maybe it's a bit cleaner to keep the `declaration` in the separate file, say `types.d.ts`

**Review by andrii-lysenko — DISMISSED** [review]
Thanks for working on this. This does seem reasonable to me. While testing on different envs haven't found any problems. Looks good 👍 
Left a small notice but it's not blocking.

File: projects/js-packages/components/tools/get-site-admin-url/index.ts (hunk 2 above)
Comment (darssen) [reply projects/js-packages/components/tools/get-site-admin-url/index.ts]: Thanks for the suggestion, I have updated it.

**Review by andrii-lysenko — APPROVED** [review]
Thanks for addressing that. LGTM!

**Pull request merged**